vsintegration/src/FSharp.Editor/QuickInfoProvider.fs

```diff
@@ -0,0 +1,98 @@
+    interface IDeferredQuickInfoContent with
+        override this.Create() : FrameworkElement =
+            let label = new Label()
+            label.Content <- content
+            label.Foreground <- SolidColorBrush(Colors.White)
+            upcast label
+
+[<Shared>]
+[<ExportQuickInfoProvider(PredefinedQuickInfoProviderNames.Semantic, FSharpCommonConstants.FSharpLanguageName)>]
+type internal FSharpQuickInfoProvider [<ImportingConstructor>] (serviceProvider: SVsServiceProvider) =
+
+    let xmlMemberIndexService = serviceProvider.GetService(typeof<SVsXMLMemberIndexService>) :?> IVsXMLMemberIndexService
+    let documentationBuilder = XmlDocumentation.CreateDocumentationBuilder(xmlMemberIndexService, serviceProvider.DTE)
+
+    static member ProvideQuickInfo(sourceText: SourceText, position: int, options: FSharpProjectOptions, filePath: string, textVersionHash: int) = async {
+        let! parseResults = FSharpChecker.Instance.ParseFileInProject(filePath, sourceText.ToString(), options)
+        let! checkFileAnswer = FSharpChecker.Instance.CheckFileInProject(parseResults, filePath, textVersionHash, sourceText.ToString(), options)
+        let checkFileResults = match checkFileAnswer with
+                               | FSharpCheckFileAnswer.Aborted -> failwith "Compilation isn't complete yet"
+                               | FSharpCheckFileAnswer.Succeeded(results) -> results
+
+        let textLine = sourceText.Lines.GetLineFromPosition(position)
+        let textLineNumber = textLine.LineNumber + 1 // Roslyn line numbers are zero-based
+        let qualifyingNames, partialName = QuickParse.GetPartialLongNameEx(textLine.ToString(), position - textLine.Start - 1)
+        let! toolTipElement = checkFileResults.GetToolTipTextAlternate(textLineNumber, position, textLine.ToString(), qualifyingNames, tagOfToken(token.IDENT(partialName)))
+
+        return toolTipElement
+    }
+    
+    interface IQuickInfoProvider with
+        override this.GetItemAsync(document: Document, position: int, cancellationToken: CancellationToken): Task<QuickInfoItem> =
+            let computation = async {
+                match FSharpLanguageService.GetOptions(document.Project.Id) with
+                | Some(options) ->
+                    let! sourceText = document.GetTextAsync(cancellationToken) |> Async.AwaitTask
+                    let textSpan = TextSpan(0, sourceText.Length)
+                    let defines = CompilerEnvironment.GetCompilationDefinesForEditing(document.Name, options.OtherOptions |> Seq.toList)
+                    let tokens = FSharpColorizationService.GetColorizationData(document.Id, sourceText, textSpan, Some(document.FilePath), defines, cancellationToken)
+
+                    match tokens |> Seq.tryFind(fun t -> t.TextSpan.Contains(position)) with
+                    | Some(token) ->
+                        match token.ClassificationType with
+                        | ClassificationTypeNames.Identifier ->
```

Comment:
In the GotoDefinition implementation we now have 

```
                match classifiedSpan.ClassificationType with
                | ClassificationTypeNames.ClassName
                | ClassificationTypeNames.DelegateName
                | ClassificationTypeNames.EnumName
                | ClassificationTypeNames.InterfaceName
                | ClassificationTypeNames.ModuleName
                | ClassificationTypeNames.StructName
                | ClassificationTypeNames.TypeParameterName
                | ClassificationTypeNames.Identifier -> 
```
So I'm assuming quickinfo won't work at operators with this PR.

I think you should abstract out the ``tryClassifyAtPosition`` implementation and use it in both places.